File: vb/word_reverser/word_reverser_9.vb
' This program is called Word Reverser.
' The purpose of this program is to take a word input by the user and reverse the characters in the word.
' The reversed word is then displayed to the user.
' This program is designed to help programmers understand how to manipulate strings in Visual Basic.

Module WordReverser

    Sub Main()
        ' Declare a variable to hold the user's input
        Dim userInput As String

        ' Prompt the user to enter a word
        Console.WriteLine("Please enter a word to reverse:")

        ' Read the user's input from the console
        userInput = Console.ReadLine()

        ' Call the function to reverse the word
        Dim reversedWord As String = ReverseWord(userInput)

        ' Display the reversed word to the user
        Console.WriteLine("The reversed word is: " & reversedWord)

        ' Wait for the user to press a key before closing the console window
        Console.ReadKey()
    End Sub

    ' This function takes a word as input and returns the word with its characters reversed
    Function ReverseWord(ByVal word As String) As String
        ' Declare a variable to hold the reversed word
        Dim reversed As String = ""

        ' Loop through each character in the word starting from the end
        For i As Integer = word.Length - 1 To 0 Step -1
            ' Add each character to the reversed word
            reversed &= word.Chars(i)
        Next

        ' Return the reversed word
        Return reversed
    End Function

End Module

File: csharp/word_reverser/word_reverser_9.cs
#include <stdio.h>
#include <string.h>

// This program is called "word reverser". It takes a single word as input from the user
// and prints the word in reverse order. The purpose of this program is to help programmers
// understand how to manipulate strings in C and practice debugging skills. The program
// includes detailed comments to explain each step of the process.

void reverseWord(char word[]) {
    int length = strlen(word); // Get the length of the input word
    char reversed[length + 1]; // Create a new array to hold the reversed word, including the null terminator

    // Loop through the input word from the end to the beginning
    for (int i = 0; i < length; i++) {
        reversed[i] = word[length - i - 1]; // Copy characters from the end of the input word to the beginning of the reversed word
    }

    reversed[length] = '\0'; // Add the null terminator to the end of the reversed word

    printf("Reversed word: %s\n", reversed); // Print the reversed word
}

int main() {
    char word[100]; // Create an array to hold the input word

    printf("Enter a word: "); // Prompt the user to enter a word
    scanf("%s", word); // Read the input word from the user

    reverseWord(word); // Call the function to reverse the word

    return 0; // Return 0 to indicate successful execution
}

